sdk/lib/rtl/heap.c

```diff
@@ -1089,6 +1105,9 @@ RtlpInitializeHeapSegment(IN OUT PHEAP Heap,
                     FreeEntry->SegmentOffset = SegmentIndex;
                     FreeEntry->Size = GuardEntry->PreviousSize;
```

Comment:
So, if I understand correctly now, this `FreeEntry->Size` is going to use the "old" `GuardEntry->PreviousSize` ?

Reply:
From Above:

> It no longer fixed Process Hacker for me. I have now updated it with info from @ThFabba's PR https://github.com/reactos/reactos/pull/4915 and it now fixes Process Hacker again. I only used the most basic part of his PR. There is still much to gain with what he has in his PR.

My intent was not to "copy" Thomas Faber's work, but to only "borrow" the minimum from it for now. I agree that what he has is more "robust", but this should be left to him and the PR https://github.com/reactos/reactos/pull/4915. Thanks.

Reply:
I have now changed the code to match what Thomas Faber was doing in his PR #4915 as you pointed to his code above.